crates/autopilot/src/domain/settlement/solution.rs

```diff
@@ -0,0 +1,363 @@
+                tokens.iter().position(|token| token == &buy_token).unwrap();
+            trades.push(trade::Trade::new(
+                tokenized::order_uid(&trade, &tokens, domain_separator)
+                    .map_err(|err| error::Encoding::OrderUidRecover(err).with(auction_id))?,
+                eth::Asset {
+                    token: sell_token.into(),
+                    amount: trade.3.into(),
+                },
+                eth::Asset {
+                    token: buy_token.into(),
+                    amount: trade.4.into(),
+                },
+                flags.side(),
+                trade.9.into(),
+                trade::Prices {
+                    uniform: trade::ClearingPrices {
+                        sell: clearing_prices[uniform_sell_token_index],
+                        buy: clearing_prices[uniform_buy_token_index],
+                    },
+                    custom: trade::ClearingPrices {
+                        sell: clearing_prices[sell_token_index],
+                        buy: clearing_prices[buy_token_index],
+                    },
+                },
+            ));
+        }
+
+        Ok(Self { trades, auction_id })
+    }
+}
+
+/// Trade flags are encoded in a 256-bit integer field. For more information on
+/// how flags are encoded see:
+/// <https://github.com/cowprotocol/contracts/blob/v1.0.0/src/contracts/libraries/GPv2Trade.sol#L58-L94>
+#[derive(Debug, PartialEq, Eq)]
+pub struct TradeFlags(pub U256);
+
+impl TradeFlags {
+    fn as_u8(&self) -> u8 {
+        self.0.byte(0)
+    }
+
+    pub fn side(&self) -> order::Side {
+        if self.as_u8() & 0b1 == 0 {
+            order::Side::Sell
+        } else {
+            order::Side::Buy
+        }
+    }
+
+    pub fn partially_fillable(&self) -> bool {
+        self.as_u8() & 0b10 != 0
+    }
+
+    pub fn sell_token_balance(&self) -> boundary::SellTokenSource {
+        if self.as_u8() & 0x08 == 0 {
+            boundary::SellTokenSource::Erc20
+        } else if self.as_u8() & 0x04 == 0 {
+            boundary::SellTokenSource::External
+        } else {
+            boundary::SellTokenSource::Internal
+        }
+    }
+
+    pub fn buy_token_balance(&self) -> boundary::BuyTokenDestination {
+        if self.as_u8() & 0x10 == 0 {
+            boundary::BuyTokenDestination::Erc20
+        } else {
+            boundary::BuyTokenDestination::Internal
+        }
+    }
+
+    pub fn signing_scheme(&self) -> boundary::SigningScheme {
+        match (self.as_u8() >> 5) & 0b11 {
+            0b00 => boundary::SigningScheme::Eip712,
+            0b01 => boundary::SigningScheme::EthSign,
+            0b10 => boundary::SigningScheme::Eip1271,
+            0b11 => boundary::SigningScheme::PreSign,
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl From<U256> for TradeFlags {
+    fn from(value: U256) -> Self {
+        Self(value)
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
```

Comment:
Still exports `Error` type and `error` module publicly. Is there a good reason to do so?

Reply:
changed with the refactor